partiql-ast/src/main/java/org/partiql/ast/DataType.java

```diff
@@ -74,6 +71,30 @@ public List<AstNode> getChildren() {
         public <R, C> R accept(@NotNull AstVisitor<R, C> visitor, C ctx) {
             return visitor.visitStructField(this, ctx);
         }
+
+        @NotNull
```

Comment:
(self-review) left out the javadoc for most of the
- getters
- static enum constructor methods
- default constructors

imo, those functions and constructor arguments are pretty self-explanatory.